File: 041- Maximum Favourite Songs in Playlist/MaxFavouriteSong.py
def max_favourite_songs(S: str, K: int) -> int:
    count = S[:K].count('a')
    max_count = count

    for i in range(K, len(S)):
        if S[i - K] == 'a':
            count -= 1
        if S[i] == 'a':
            count += 1
        max_count = max(max_count, count)

    return max_count

# Example usage
S = "acdbaaca"
K = 3
print(max_favourite_songs(S, K))  # Output: 2

File: 041- Maximum Favourite Songs in Playlist/MaxFavouriteSong.cpp
#include<iostream>
#include<string>
using namespace std;

int maxFavouriteSongs(string s, int k){
    int maxcount = 0;
    int Count = 0;

    for(int i = 0; i < k; i++){
        if(s[i] == 'a'){
            Count++;
        }
    }
    maxcount = Count;

    for(int i = k; i < s.length(); i++){
        if(s[i - k] == 'a'){
            Count--;
        }
        if(s[i] == 'a'){
            Count++;
        }
        maxcount = max(maxcount, Count);
    }
    return maxcount;
}
int main() {
    string S = "acdbaaca";
    int K = 3;
    cout << maxFavouriteSongs(S, K) << endl; // Output: 2
    return 0;
}